ra/ra.go

```diff
@@ -880,7 +877,8 @@ func mergeUpdate(r *core.Registration, input core.Registration) bool {
 		changed = true
 	}
 
-	if features.Enabled(features.AllowKeyRollover) && input.Key != emptyKey && input.Key != r.Key {
+	if features.Enabled(features.AllowKeyRollover) &&
```

Comment:
I can take a crack at that and see how much it will ripple out through other places.